src/tests/Common/CLRTest.Execute.Bash.targets

```diff
@@ -274,7 +274,7 @@ else
   __Command+=" dotnet"
 fi
 
-$__Command msbuild $CORE_ROOT/wasm-test-runner/WasmTestRunner.proj /p:NetCoreAppCurrent=$(NetCoreAppCurrent) /p:TestAssembly=`pwd`/$(MsBuildProjectName).dll /p:TestBinDir=`pwd`
+$__Command msbuild $CORE_ROOT/wasm-test-runner/WasmTestRunner.proj /p:NetCoreAppCurrent=$(NetCoreAppCurrent) /p:TestAssembly=`pwd`/$(MsBuildProjectName).dll /p:TestBinDir=`pwd` || exit $?
```

Comment:
This is how I handled the error of this script. I think it is better this way, because it covers the whole script.
https://github.com/dotnet/runtime/blob/master/src/tests/Common/CLRTest.Execute.Bash.targets#L409-L413
You could add the WASM condition there.

Reply:
True, that would be better, but I wasn't sure what that might break lol. @naricc thoughts?

Reply:
We can maybe merge this one, and then explore the `set -e` in a separate PR

Reply:
I think using set -e  works for android because the tests use the xharness runner which is returning 0 if the test succeeds. Wasm is using the return code the test assemblies themselves produce, which is 100 on success, but looks like an error to bash.

However, putting it as the first part of an || statement like in your PR would keep bash from seeing is as an error that requires termination with `set -e`.

So, regardless of whether or not we use `set -e`, we need this change.

Reply:
Then `set -e` would cause it to exit early. We probably shouldn't use it for WASM.